client/webserver/site/src/js/markets.ts

```diff
@@ -604,9 +604,74 @@ export default class MarketsPage extends BasePage {
   /* assetsAreSupported is true if all the assets of the current market are
    * supported
    */
-  assetsAreSupported () {
-    const [b, q] = [this.market.base, this.market.quote]
-    return b && q
+  assetsAreSupported (): {
+    isSupported: boolean;
+    text: string;
+    } {
+    const { market } = this
+    const [base, quote] = [market.base, market.quote]
+    if (!base || !quote) {
+      const symbol = market.base ? market.quoteCfg.symbol : market.baseCfg.symbol
+      return {
+        isSupported: false,
+        text: intl.prep(intl.ID_NOT_SUPPORTED, { asset: symbol.toUpperCase() })
+      }
+    }
+    const { baseCfg, quoteCfg } = this.market
+    // check if versions are supported. If asset is a token, we check if its
+    // parent supports the version.
+    let text = ''
+    if (base.token && quote.token) {
+      const bParent = app().assets[base.token.parentID]
+      const qParent = app().assets[quote.token.parentID]
+      if (!bParent.info?.versions.includes(baseCfg.version)) {
```

Comment:
I don't think this is correct, you're checking the parent's version against the token version. The token also has its own version independent of the parent asset, but it doesn't seem to be exposed through the `SupportedAsset` type. I think the token wallet's version also needs to be exposed through the `SupportedAsset` type. @buck54321  Am I correct about this?

Reply:
Sorry for the late response. Yes. We are not properly exposing supported versions for tokens. @vctt94, check out https://github.com/buck54321/dcrdex/compare/97a8cb2593588862c0ac7774f83fbe1e84dae952...buck54321:dcrdex:supported-versions-addon

Reply:
Just opened #2094 too, so if that goes in first, you can just rebase to get the client/asset parts.

Reply:
I've changed tacts on this. See the updated #2094, but I think we'll have the server maintain a single protocol version and token assets will share both a version and supported versions with the parent asset. The code here is roughly correct, though I'd recommend the following refactor https://github.com/buck54321/dcrdex/compare/97a8cb2593588862c0ac7774f83fbe1e84dae952...buck54321:dcrdex:parent-vers-pattern

The only difference is that it will show the token's symbol instead of the parent asset, but I think that makes sense anyway.